packages/stripe-integration/src/stripe-upe/stripe-upe-integration-service.ts

```diff
@@ -0,0 +1,222 @@
+import { includes, some } from 'lodash';
+
+import {
+    Address,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType,
+    PaymentIntegrationService,
+    PaymentMethodFailedError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import {
+    AddressOptions,
+    StripeAdditionalActionRequired,
+    StripeConfirmPaymentData,
+    StripeElement,
+    StripeElements,
+    StripeElementType,
+    StripeError,
+    StripeStringConstants,
+    StripeUPEClient,
+    StripeUPEPaymentIntentStatus,
+} from './stripe-upe';
+import StripeUPEPaymentInitializeOptions from './stripe-upe-initialize-options';
+
+export default class StripeUPEIntegrationService {
+    private isMounted = false;
+    private checkoutEventsUnsubscribe?: () => void;
+
+    constructor(private paymentIntegrationService: PaymentIntegrationService) {}
+
+    deinitialize(): void {
+        this.checkoutEventsUnsubscribe?.();
+        this.isMounted = false;
+    }
+
+    initCheckoutEventsSubscription(
+        gatewayId: string,
+        methodId: string,
+        stripeupe: StripeUPEPaymentInitializeOptions,
+        stripeElements?: StripeElements,
+    ): void {
+        this.checkoutEventsUnsubscribe = this.paymentIntegrationService.subscribe(
+            async () => {
+                const paymentElement = stripeElements?.getElement(StripeElementType.PAYMENT);
+
+                if (!paymentElement) {
+                    return;
+                }
+
+                let error;
+
+                try {
+                    await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
+                        params: { method: methodId },
+                    });
+                } catch (err) {
+                    error = err;
+                }
+
+                if (error && error instanceof Error) {
+                    if (this.isMounted) {
+                        paymentElement.unmount();
+                        this.isMounted = false;
+                    }
+
+                    stripeupe.onError?.(error);
+                } else if (!this.isMounted) {
+                    await stripeElements?.fetchUpdates();
+                    this.mountElement(paymentElement, stripeupe.containerId);
+                }
+            },
```

Comment:
[optional] This method can be simplified:

```suggestion
            async () => {
                const paymentElement = stripeElements?.getElement(StripeElementType.PAYMENT);

                if (!paymentElement) {
                    return;
                }

                try {
                    await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
                        params: { method: methodId },
                    });
                } catch (error) {
                   if (this.isMounted) {
                        paymentElement.unmount();
                        this.isMounted = false;
                    }

                    stripeupe.onError?.(error);
                    
                    return;
                }

                if (!this.isMounted) {
                    await stripeElements?.fetchUpdates();
                    this.mountElement(paymentElement, stripeupe.containerId);
                }
            },
```